evm/src/cross_table_lookup.rs

```diff
@@ -21,42 +23,127 @@ use crate::proof::{StarkProofWithPublicInputs, StarkProofWithPublicInputsTarget}
 use crate::stark::Stark;
 use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
 
+/// Represent a linear combination of columns.
 #[derive(Clone)]
-pub struct TableWithColumns {
-    table: Table,
-    columns: Vec<usize>,
-    /// Vector of columns `[c_1,...,c_k]` used as a filter using the sum `c_1 + ... + c_k`.
-    /// An empty vector corresponds to no filter.
-    filter_columns: Vec<usize>,
+pub struct Column<F: Field> {
+    linear_combination: Vec<(usize, F)>,
+    constant: F,
 }
 
-impl TableWithColumns {
-    pub fn new(table: Table, columns: Vec<usize>, filter_columns: Vec<usize>) -> Self {
+impl<F: Field> Column<F> {
+    pub fn single(c: usize) -> Self {
+        Self {
+            linear_combination: vec![(c, F::ONE)],
+            constant: F::ZERO,
+        }
+    }
+
+    pub fn singles(cs: Vec<usize>) -> Vec<Self> {
+        cs.into_iter().map(Self::single).collect()
+    }
+
+    pub fn linear_combination_with_constant<I: IntoIterator<Item = (usize, F)>>(
+        iter: I,
+        constant: F,
+    ) -> Self {
+        let v = iter.into_iter().collect::<Vec<_>>();
+        assert!(!v.is_empty());
         debug_assert_eq!(
-            filter_columns.iter().unique().count(),
-            filter_columns.len(),
-            "Duplicate filter columns."
+            v.iter().map(|(c, _)| c).unique().count(),
+            v.len(),
+            "Duplicate columns."
         );
+        Self {
+            linear_combination: v,
+            constant,
+        }
+    }
+
+    pub fn linear_combination<I: IntoIterator<Item = (usize, F)>>(iter: I) -> Self {
+        Self::linear_combination_with_constant(iter, F::ZERO)
+    }
+
+    pub fn le_bits<I: IntoIterator<Item = usize>>(cs: I) -> Self {
+        Self::linear_combination(cs.into_iter().zip(F::TWO.powers()))
+    }
+
+    pub fn sum(cs: &[usize]) -> Self {
```

Comment:
Nit: How about `pub fn sum<I: IntoIterator<Item = usize>>(cs: I) -> Self {`? Would be slightly more flexible.